devsetup/scripts/gen-edpm-compute-node.sh

```diff
@@ -39,6 +39,10 @@ EDPM_COMPUTE_NETWORK_TYPE=${EDPM_COMPUTE_NETWORK_TYPE:-network}
 # '[{"type": "network", "name": "crc-bmaas"}, {"type": "network", "name": "other-net"}]'
 EDPM_COMPUTE_ADDITIONAL_NETWORKS=${2:-'[]'}
 EDPM_COMPUTE_NETWORK_IP=$(virsh net-dumpxml ${EDPM_COMPUTE_NETWORK} | xmllint - --xpath 'string(/network/ip/@address)')
+VALID_IP_REGEX="^([0-9]{1,3}\.){3}[0-9]{1,3}$"
```

Comment:
I've used xmllint to grab other values from libvirt elsewhere so I worry this can be a problem in other places as well ...

The output in pastebin looks very strange.
```
  nmcli device modify eth0 ipv4.addresses <?xml version="1.0"?>             <------------------------------------------ bad
<network connections="1">
  <name>default</name>
  <uuid>c553c526-7a57-45ce-b897-94a416f056c4</uuid>
  <forward mode="nat">
    <nat>
      <port start="1024" end="65535"/>
    </nat>
  </forward>
  <bridge name="virbr0" stp="on" delay="0"/>
  <mtu size="9000"/>
  <mac address="52:54:00:c0:57:63"/>
  <ip address="192.168.122.1" netmask="255.255.255.0">
    <dhcp>
      <range start="192.168.122.2" end="192.168.122.254"/>
      <host mac="52:54:00:69:AE:74" name="crc" ip="192.168.122.100/24 ipv4.gateway <?xml version="1.0"?>
```
 Here in the middle of xml the string `ipv4.gateway` which is likely part of the nmcli command?

```<network connections="1">
  <name>default</name>
  <uuid>c553c526-7a57-45ce-b897-94a416f056c4</uuid>
  <forward mode="nat">
    <nat>
      <port start="1024" end="65535"/>
    </nat>
  </forward>
  <bridge name="virbr0" stp="on" delay="0"/>
  <mtu size="9000"/>
  <mac address="52:54:00:c0:57:63"/>
  <ip address="192.168.122.1" netmask="255.255.255.0">
    <dhcp>
      <range start="192.168.122.2" end="192.168.122.254"/>
      <host mac="52:54:00:69:AE:74" name="crc" ip="192.168.122.10"/>
    </dhcp>
  </ip>
</network> ipv4.dns <?xml version="1.0"?>
```
Again, `ipv4.dns` in the middle of XML which is also part of the nmcli command?

```
<network connections="1">
```

What does `virsh net-dumpxml default` return on RHEL8?
And what if you pipe it ti xmllint `| xmllint - --xpath 'string(/network/ip/@address)` ?

Reply:
Got it, we need to put the `-` to tell xmllint to read from stdin as the last argument on RHEL8.

`┻━┻︵ \(°□°)/ ︵ ┻━┻`

```
[root@rhel8 ~]# virsh net-dumpxml default | xmllint --xpath 'string(/network/ip/@address)' -
192.168.124.1[root@rhel8 ~]# 
```

```
[root@rhel8 ~]# virsh net-dumpxml default | xmllint - --xpath 'string(/network/ip/@address)'
<?xml version="1.0"?>
<network>
  <name>default</name>
  <uuid>9fd99cfc-c2d9-4bc8-ad70-3ba13bc9d1c6</uuid>
  <forward mode="nat">
    <nat>
      <port start="1024" end="65535"/>
    </nat>
  </forward>
  <bridge name="virbr0" stp="on" delay="0"/>
  <mac address="52:54:00:15:9d:43"/>
  <ip address="192.168.124.1" netmask="255.255.255.0">
    <dhcp>
      <range start="192.168.124.2" end="192.168.124.254"/>
    </dhcp>
  </ip>
</network>
```